Adds nupic.proto to bindings installation.

scottpurdy opened the pull request:
fixes #563 

This means nupic will no longer need to copy files over and the proto files will be importable from where they are installed as part of bindings installation (nupic.proto.*).

This also switches the imports to be absolute which we will need to change in nupic as well. I tested this locally and it appears to work fine, both for development and regular installs.

@oxtopus please review
@akhilaananthram fyi

oxtopus commented:
May need to add "nupic.proto" to https://github.com/scottpurdy/nupic.core/blob/nupic.proto/bindings/py/setup.py#L565 and follow rest of namespace package convention.

oxtopus commented:
I doubt it's related, but I get a nupic test failure when I test against this version:

```
tests/unit/nupic/encoders/coordinate_test.py:202: CoordinateEncoderTest.testEncodeRelativePositionsAndRadii FAILED

========================================================================================================================================================== FAILURES ==========================================================================================================================================================
_________________________________________________________________________________________________________________________________ CoordinateEncoderTest.testEncodeRelativePositionsAndRadii __________________________________________________________________________________________________________________________________
self = <tests.unit.nupic.encoders.coordinate_test.CoordinateEncoderTest testMethod=testEncodeRelativePositionsAndRadii>

    def testEncodeRelativePositionsAndRadii(self):
      # As radius increases and positions change, the overlap should decrease
      overlaps = overlapsForRelativeAreas(999, 25, np.array([100, 200]), 5,
                                          dPosition=np.array([1, 1]),
                                          dRadius=1,
                                          num=5)
>     self.assertDecreasingOverlaps(overlaps)

tests/unit/nupic/encoders/coordinate_test.py:208: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <tests.unit.nupic.encoders.coordinate_test.CoordinateEncoderTest testMethod=testEncodeRelativePositionsAndRadii>, overlaps = array([ 0.64,  0.44,  0.2 ,  0.2 ,  0.16])

    def assertDecreasingOverlaps(self, overlaps):
>     self.assertEqual((np.diff(overlaps) >= 0).sum(), 0)
E     AssertionError: 1 != 0

tests/unit/nupic/encoders/coordinate_test.py:267: AssertionError
------------------------------------------------------------------------------------------------------------------------------------------------------ Captured stderr -------------------------------------------------------------------------------------------------------------------------------------------------------
/Users/amarshall/nta/nupic/tests/unit/nupic/encoders/coordinate_test.py:330: FutureWarning: comparison to `None` will result in an elementwise object comparison in the future.
  newPosition = initPosition if dPosition == None else (
!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!! Interrupted: stopping after 1 failures !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
====================================================================================================================================== 1 failed, 196 passed, 9 skipped in 6.79 seconds =======================================================================================================================================
```

scottpurdy commented:
@oxtopus - I believe that having `nupic` be a namespace package is what is needed. `nupic.proto` is inside that namespace package but is not itself a namespace package.

Currently `nupic` is a namespace package and `nupic.bindings` is also one but only because @akhilaananthram had to put the nupic.core proto files into a sub-package of `nupic.bindings` during the nupic build. With this change, `nupic.bindings` will no longer be a namespace package.

Regarding that test failure, I don't see it locally. I will have to update the SHA in nupic and run travis so if the issue is really there I will address it then.

oxtopus commented:
The change you describe re:

> With this change, nupic.bindings will no longer be a namespace package.

Is not conveyed in this PR.

oxtopus commented:
I created https://github.com/numenta/nupic/issues/2471 to track the test failure, which happens for me in current nupic@master and nupic.core@master and not introduced by this change.

scottpurdy commented:
@oxtopus - Good point. Removed `nupic.bindings` from namespace packages and dropped the inclusion of .capnp files from distribution archive in .travis.yml. Good now?

scottpurdy commented:
@oxtopus - ping

scottpurdy commented:
@oxtopus - Had to resolve some conflicts - please take a look when you can. This is ready for merge AFAIK

oxtopus commented:
:+1:
